test/sys/test_aio_drop.rs

```diff
@@ -1,6 +1,11 @@
 extern crate nix;
 extern crate tempfile;
 
+#[cfg(any(target_os = "linux",
+          target_os = "ios",
+          target_os = "macos",
+          target_os = "freebsd",
+          target_os = "netbsd"))]
 use nix::sys::aio::*;
```

Comment:
The build of this file doubtlessly failed on OpenBSD because the entire aio module is not included.  But by disabling it in this way, you'll probably get "unused import" warnings for signal, AsRawFd, and tempfile.  If so, can you eliminate the warnings?  Since there's only one test case in this file, one way to eliminate the warnings would be to move them into the test method itself.

Reply:
@asomers thanks for your comments, you are right of course, there where "unused import" warnings on OpenBSD.

I've moved the imports inside the test function. I've ran the test suite on OpenBSD 6.4 (where AIO is not supported) and on FreeBSD 12 (where AIO works) and everything if fine (except for `sys::test_socket::test_scm_rights` which was failing on OpenBSD from the very begining). Long outputs follow.

#### OpenBSD 6.4
```bash
user1@openbsd64 ~/ALACRITTY/nix $ cargo clean

user1@openbsd64 ~/ALACRITTY/nix $ cargo build
   Compiling libc v0.2.45 (https://github.com/rust-lang/libc/#027d4834)
   Compiling nix v0.12.0 (/home/user1/ALACRITTY/nix)
   Compiling cfg-if v0.1.6
   Compiling void v1.0.2
   Compiling bitflags v1.0.4
    Finished dev [unoptimized + debuginfo] target(s) in 14.90s

user1@openbsd64 ~/ALACRITTY/nix $ cargo build --release
   Compiling libc v0.2.45 (https://github.com/rust-lang/libc/#027d4834)
   Compiling nix v0.12.0 (/home/user1/ALACRITTY/nix)
   Compiling cfg-if v0.1.6
   Compiling void v1.0.2
   Compiling bitflags v1.0.4
    Finished release [optimized] target(s) in 13.20s

user1@openbsd64 ~/ALACRITTY/nix $ cargo test
   Compiling semver-parser v0.7.0
   Compiling libc v0.2.45
   Compiling rand_core v0.3.0
   Compiling byteorder v1.2.7
   Compiling remove_dir_all v0.5.1
   Compiling lazy_static v1.2.0
   Compiling rand_isaac v0.1.1
   Compiling rand_xorshift v0.1.0
   Compiling rand_hc v0.1.0
   Compiling rand_core v0.2.2
   Compiling semver v0.9.0
   Compiling iovec v0.1.2
   Compiling rand v0.5.5
   Compiling bytes v0.4.11
   Compiling rustc_version v0.2.3
   Compiling rand_chacha v0.1.0
   Compiling rand_pcg v0.1.1
   Compiling rand v0.6.1
   Compiling tempfile v3.0.5
   Compiling nix v0.12.0 (/home/user1/ALACRITTY/nix)
    Finished dev [unoptimized + debuginfo] target(s) in 20.76s
     Running target/debug/deps/nix-99a4ce04750d6005

running 35 tests
test sys::event::test_struct_kevent ... ok
test ifaddrs::tests::test_getifaddrs ... ok
test sys::select::tests::fdset_clear ... ok
test sys::select::tests::test_select ... ok
test sys::select::tests::fdset_remove ... ok
test sys::select::tests::fdset_highest ... ok
test sys::select::tests::fdset_insert ... ok
test sys::select::tests::test_select_nfds ... ok
test sys::signal::tests::test_clear ... ok
test sys::signal::tests::test_contains ... ok
test sys::select::tests::test_select_nfds2 ... ok
test sys::signal::tests::test_from_str_invalid_value ... ok
test sys::signal::tests::test_extend ... ok
test sys::signal::tests::test_from_str_round_trips ... ok
test sys::signal::tests::test_sigaction ... ok
test sys::signal::tests::test_sigwait ... ok
test sys::signal::tests::test_thread_signal_get_mask ... ok
test sys::signal::tests::test_thread_signal_block ... ok
test sys::signal::tests::test_thread_signal_set_mask ... ok
test sys::signal::tests::test_thread_signal_swap ... ok
test sys::signal::tests::test_thread_signal_unblock ... ok
test sys::socket::addr::tests::test_macos_tap_datalink_addr ... ok
test sys::socket::sockopt::test::is_socket_type_dgram ... ok
test sys::socket::addr::tests::test_macos_loopback_datalink_addr ... ok
test sys::statvfs::test::statvfs_call ... ok
test sys::statvfs::test::fstatvfs_call ... ok
test sys::socket::sockopt::test::is_socket_type_unix ... ok
test sys::time::test::test_timespec_ord ... ok
test sys::time::test::test_timeval_neg ... ok
test sys::time::test::test_timespec ... ok
test sys::time::test::test_timespec_neg ... ok
test sys::time::test::test_timeval ... ok
test sys::time::test::test_timeval_fmt ... ok
test sys::time::test::test_timespec_fmt ... ok
test sys::time::test::test_timeval_ord ... ok

test result: ok. 35 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out

     Running target/debug/deps/test-58f96996c19b993c

running 88 tests
test sys::test_ioctl::bsd::test_op_write_64 ... ok
test sys::test_ioctl::bsd::test_op_read_write ... ok
test sys::test_ioctl::bsd::test_op_read ... ok
test sys::test_ioctl::bsd::test_op_none ... ok
test sys::test_ioctl::bsd::test_op_read_64 ... ok
test sys::test_ioctl::bsd::test_op_write ... ok
test sys::test_ioctl::bsd::test_op_read_write_64 ... ok
test sys::test_ptrace::test_ptrace ... ok
test sys::test_signal::test_old_sigaction_flags ... ok
test sys::test_select::test_pselect_nfds2 ... ok
test sys::test_select::test_pselect ... ok
test sys::test_pthread::test_pthread_self ... ok
test sys::test_signal::test_kill_none ... ok
test sys::test_ptrace::test_ptrace_cont ... ok
test sys::test_signal::test_sigprocmask ... ok
test sys::test_signal::test_sigprocmask_noop ... ok
test sys::test_socket::test_inetv4_addr_to_sock_addr ... ok
test sys::test_socket::test_inetv6_addr_to_sock_addr ... ok
test sys::test_socket::test_path_to_sock_addr ... ok
test sys::test_socket::test_socketpair ... ok
test sys::test_socket::test_scm_rights ... FAILED
test sys::test_socket::test_sendmsg_empty_cmsgs ... ok
test sys::test_socket::test_getsockname ... ok
test sys::test_termios::test_output_flags ... ok
test sys::test_termios::test_cfmakeraw ... ok
test sys::test_socket::test_scm_rights_single_cmsg_multiple_fds ... ok
test sys::test_sockopt::test_so_buf ... ok
test sys::test_termios::test_local_flags ... ok
test sys::test_termios::test_tcgetattr_ebadf ... ok
test sys::test_termios::test_tcgetattr_enotty ... ok
test sys::test_termios::test_tcgetattr_pty ... ok
test sys::test_uio::test_pwrite ... ok
test sys::test_uio::test_readv ... ok
test sys::test_wait::test_wait_exit ... ok
test sys::test_uio::test_pread ... ok
test sys::test_wait::test_waitstatus_from_raw ... ok
test sys::test_socket::test_unixdomain ... ok
test sys::test_wait::test_wait_signal ... ok
test sys::test_uio::test_writev ... ok
test test_dir::ebadf ... ok
test sys::test_wait::test_waitstatus_pid ... ok
test test_poll::test_poll_debug ... ok
test test_net::test_if_nametoindex ... ok
test test_fcntl::test_openat ... ok
test test_pty::test_open_ptty_pair ... ok
test test_fcntl::test_readlink ... ok
test test_pty::test_openpty ... ok
test test_pty::test_explicit_close ... ok
test test_dir::read ... ok
test test_dir::rewind ... ok
test test_pty::test_openpty_with_termios ... ok
test test_stat::test_fchmod ... ok
test test_stat::test_futimens ... ok
test test_stat::test_fchmodat ... ok
test_acct requires root privileges. Skipping test.
test test_stat::test_fstatat ... ok
test test_stat::test_stat_and_fstat ... ok
test test_stat::test_utimensat ... ok
test test_unistd::test_acct ... ok
test test_stat::test_stat_fstat_lstat ... ok
test test_stat::test_utimes ... ok
test test_unistd::test_fchownat ... ok
test test_unistd::test_chown ... ok
test test_unistd::test_fchdir ... ok
test test_unistd::test_execve ... ok
test test_unistd::test_fpathconf_limited ... ok
test test_unistd::test_ftruncate ... ok
test test_unistd::test_getpid ... ok
test test_unistd::test_getsid ... ok
test_initgroups requires root privileges. Skipping test.
test test_unistd::test_initgroups ... ok
test test_unistd::test_lseek ... ok
test test_unistd::test_mkfifo_directory ... ok
test test_unistd::test_mkfifo ... ok
test test_unistd::test_execvpe ... ok
test test_unistd::test_fork_and_waitpid ... ok
test test_unistd::test_mkstemp ... ok
test test_unistd::test_mkstemp_directory ... ok
test test_unistd::test_pathconf_limited ... ok
test_setgroupstest test_unistd::test_pipe ... ok
 requires root privileges. Skipping test.
test test_unistd::test_pipe2 ... ok
test test_unistd::test_sysconf_limited ... ok
test test_unistd::test_setgroups ... ok
test test_unistd::test_getcwd ... ok
test test_unistd::test_wait ... ok
test test_unistd::test_truncate ... ok
test test_unistd::test_symlinkat ... ok
test test_poll::test_poll ... ok
test test_unistd::test_alarm ... ok
test test_unistd::test_canceling_alarm ... ok

failures:

---- sys::test_socket::test_scm_rights stdout ----
thread 'sys::test_socket::test_scm_rights' panicked at 'slice index starts at 24 but ends at 20', libcore/slice/mod.rs:2340:5
note: Run with `RUST_BACKTRACE=1` for a backtrace.


failures:
    sys::test_socket::test_scm_rights

test result: FAILED. 87 passed; 1 failed; 0 ignored; 0 measured; 0 filtered out

error: test failed, to rerun pass '--test test'
```

#### FreeBSD 12
```bash
user1@freebsd12 $ cargo clean

user1@freebsd12 $ cargo build
    Updating crates.io index
    Updating git repository `https://github.com/rust-lang/libc/`
 Downloading crates ...
  Downloaded cfg-if v0.1.6
  Downloaded void v1.0.2
  Downloaded bitflags v1.0.4
   Compiling libc v0.2.45 (https://github.com/rust-lang/libc/#027d4834)
   Compiling nix v0.12.0 (/usr/home/user1/P/nix)
   Compiling bitflags v1.0.4
   Compiling void v1.0.2
   Compiling cfg-if v0.1.6
    Finished dev [unoptimized + debuginfo] target(s) in 1m 36s

user1@freebsd12 $ cargo build --release
   Compiling libc v0.2.45 (https://github.com/rust-lang/libc/#027d4834)
   Compiling nix v0.12.0 (/usr/home/user1/P/nix)
   Compiling cfg-if v0.1.6
   Compiling bitflags v1.0.4
   Compiling void v1.0.2
    Finished release [optimized] target(s) in 55.62s

user1@freebsd12 $ cargo test
 Downloading crates ...
  Downloaded lazy_static v1.2.0
  Downloaded tempfile v3.0.5
  Downloaded bytes v0.4.11
  Downloaded rand v0.5.5
  Downloaded remove_dir_all v0.5.1
  Downloaded byteorder v1.2.7
  Downloaded libc v0.2.45
  Downloaded rand v0.6.1
  Downloaded iovec v0.1.2
  Downloaded rand_core v0.2.2
  Downloaded rand_isaac v0.1.1
  Downloaded rand_xorshift v0.1.0
  Downloaded rand_chacha v0.1.0
  Downloaded rand_core v0.3.0
  Downloaded rand_pcg v0.1.1
  Downloaded rustc_version v0.2.3
  Downloaded sysctl v0.1.4
  Downloaded rand_hc v0.1.0
  Downloaded semver v0.9.0
  Downloaded semver-parser v0.7.0
  Downloaded errno v0.2.4
   Compiling semver-parser v0.7.0
   Compiling libc v0.2.45
   Compiling rand_core v0.3.0
   Compiling byteorder v1.2.7
   Compiling remove_dir_all v0.5.1
   Compiling lazy_static v1.2.0
   Compiling semver v0.9.0
   Compiling rand_isaac v0.1.1
   Compiling rand_xorshift v0.1.0
   Compiling rand_hc v0.1.0
   Compiling rand_core v0.2.2
   Compiling rustc_version v0.2.3
   Compiling rand_pcg v0.1.1
   Compiling rand_chacha v0.1.0
   Compiling rand v0.6.1
   Compiling errno v0.2.4
   Compiling iovec v0.1.2
   Compiling rand v0.5.5
   Compiling sysctl v0.1.4
   Compiling bytes v0.4.11
   Compiling tempfile v3.0.5
   Compiling nix v0.12.0 (/usr/home/user1/P/nix)
    Finished dev [unoptimized + debuginfo] target(s) in 2m 00s
     Running target/debug/deps/nix-ee4b54bf1bdafcd0

running 37 tests
test ifaddrs::tests::test_getifaddrs ... ok
test sys::event::test_struct_kevent ... ok
test sys::select::tests::fdset_clear ... ok
test sys::select::tests::fdset_highest ... ok
test sys::select::tests::fdset_insert ... ok
test sys::select::tests::test_select ... ok
test sys::select::tests::fdset_remove ... ok
test sys::select::tests::test_select_nfds ... ok
test sys::select::tests::test_select_nfds2 ... ok
test sys::signal::tests::test_clear ... ok
test sys::signal::tests::test_contains ... ok
test sys::signal::tests::test_extend ... ok
test sys::signal::tests::test_from_str_invalid_value ... ok
test sys::signal::tests::test_from_str_round_trips ... ok
test sys::signal::tests::test_sigaction ... ok
test sys::signal::tests::test_sigwait ... ok
test sys::signal::tests::test_thread_signal_get_mask ... ok
test sys::signal::tests::test_thread_signal_block ... ok
test sys::signal::tests::test_thread_signal_set_mask ... ok
test sys::signal::tests::test_thread_signal_swap ... ok
test sys::signal::tests::test_thread_signal_unblock ... ok
test sys::socket::addr::tests::test_macos_loopback_datalink_addr ... ok
test sys::socket::addr::tests::test_macos_tap_datalink_addr ... ok
test sys::socket::sockopt::test::can_get_listen_on_tcp_socket ... ok
test sys::socket::sockopt::test::is_socket_type_dgram ... ok
test sys::socket::sockopt::test::is_socket_type_unix ... ok
test sys::statvfs::test::fstatvfs_call ... ok
test sys::statvfs::test::statvfs_call ... ok
test sys::time::test::test_timespec ... ok
test sys::time::test::test_timespec_fmt ... ok
test sys::time::test::test_timespec_neg ... ok
test sys::time::test::test_timespec_ord ... ok
test sys::time::test::test_timeval ... ok
test sys::time::test::test_timeval_neg ... ok
test sys::time::test::test_timeval_fmt ... ok
test sys::time::test::test_timeval_ord ... ok
test sys::utsname::test::test_uname_freebsd ... ok

test result: ok. 37 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out

     Running target/debug/deps/test-c87e36ab2d771cee

running 123 tests
test sys::test_aio::test_accessors ... ok
test sys::test_aio::test_aio_cancel_all ... ok
test sys::test_aio::test_aio_suspend ... ok
test sys::test_aio::test_cancel ... ok
test sys::test_aio::test_fsync_error ... ok
test sys::test_aio::test_liocb_listio_nowait ... ok
test sys::test_aio::test_liocb_listio_read_immutable ... ok
test sys::test_aio::test_liocb_listio_signal ... ok
test sys::test_aio::test_liocb_listio_wait ... ok
test sys::test_aio::test_read ... ok
test sys::test_aio::test_read_bytes_mut_small ... ok
test sys::test_aio::test_read_error ... ok
test sys::test_aio::test_read_immutable_buffer ... ok
test sys::test_aio::test_read_into_mut_slice ... ok
test sys::test_aio::test_read_into_pointer ... ok
test sys::test_aio::test_write ... ok
test sys::test_aio::test_write_bytes ... ok
test sys::test_aio::test_write_error ... ok
test sys::test_aio::test_write_from_pointer ... ok
test sys::test_aio::test_write_sigev_signal ... ok
test sys::test_ioctl::bsd::test_op_none ... ok
test sys::test_ioctl::bsd::test_op_read ... ok
test sys::test_ioctl::bsd::test_op_read_64 ... ok
test sys::test_ioctl::bsd::test_op_read_write ... ok
test sys::test_ioctl::bsd::test_op_read_write_64 ... ok
test sys::test_ioctl::bsd::test_op_write ... ok
test sys::test_ioctl::bsd::test_op_write_64 ... ok
test sys::test_ioctl::bsd::test_op_write_int ... ok
test sys::test_ioctl::freebsd_ioctls::test_ioctl_none ... ok
test sys::test_ioctl::freebsd_ioctls::test_ioctl_read ... ok
test sys::test_ioctl::freebsd_ioctls::test_ioctl_write_ptr ... ok
test sys::test_pthread::test_pthread_self ... ok
test sys::test_ptrace::test_ptrace ... ok
test sys::test_ptrace::test_ptrace_cont ... ok
test sys::test_select::test_pselect ... ok
test sys::test_select::test_pselect_nfds2 ... ok
test sys::test_signal::test_kill_none ... ok
test sys::test_signal::test_old_sigaction_flags ... ok
test sys::test_signal::test_sigprocmask ... ok
test sys::test_signal::test_sigprocmask_noop ... ok
test sys::test_socket::test_getsockname ... ok
test sys::test_socket::test_inetv4_addr_to_sock_addr ... ok
test sys::test_socket::test_inetv6_addr_to_sock_addr ... ok
test sys::test_socket::test_path_to_sock_addr ... ok
test sys::test_socket::test_recv_ipv6pktinfo ... ok
test sys::test_socket::test_scm_rights ... ok
test sys::test_socket::test_scm_rights_single_cmsg_multiple_fds ... ok
test sys::test_socket::test_sendmsg_empty_cmsgs ... ok
test sys::test_socket::test_socketpair ... ok
test sys::test_socket::test_unixdomain ... ok
test sys::test_sockopt::test_so_buf ... ok
test sys::test_sockopt::test_tcp_congestion ... ok
test sys::test_termios::test_cfmakeraw ... ok
test sys::test_termios::test_local_flags ... ok
test sys::test_termios::test_output_flags ... ok
test sys::test_termios::test_tcgetattr_ebadf ... ok
test sys::test_termios::test_tcgetattr_enotty ... ok
test sys::test_termios::test_tcgetattr_pty ... ok
test sys::test_uio::test_pread ... ok
test sys::test_uio::test_pwrite ... ok
test sys::test_uio::test_readv ... ok
test sys::test_uio::test_writev ... ok
test sys::test_wait::test_wait_exit ... ok
test sys::test_wait::test_wait_signal ... ok
test sys::test_wait::test_waitstatus_from_raw ... ok
test sys::test_wait::test_waitstatus_pid ... ok
test test_dir::ebadf ... ok
test test_dir::read ... ok
test test_dir::rewind ... ok
test test_fcntl::test_openat ... ok
test test_fcntl::test_readlink ... ok
test test_mq::test_mq_getattr ... ok
test test_mq::test_mq_send_and_receive ... ok
test test_mq::test_mq_set_nonblocking ... ok
test test_mq::test_mq_setattr ... ok
test test_mq::test_mq_unlink ... ok
test test_net::test_if_nametoindex ... ok
test test_poll::test_poll ... ok
test test_poll::test_poll_debug ... ok
test test_poll::test_ppoll ... ok
test test_pty::test_explicit_close ... ok
test test_pty::test_open_ptty_pair ... ok
test test_pty::test_openpty ... ok
test test_pty::test_openpty_with_termios ... ok
test test_sendfile::test_sendfile_freebsd ... ok
test test_stat::test_fchmod ... ok
test test_stat::test_fchmodat ... ok
test test_stat::test_fstatat ... ok
test test_stat::test_futimens ... ok
test test_stat::test_lutimes ... ok
test test_stat::test_stat_and_fstat ... ok
test test_stat::test_stat_fstat_lstat ... ok
test test_stat::test_utimensat ... ok
test test_stat::test_utimes ... ok
test sys::test_aio::test_fsync ... ok
test test_unistd::test_alarm ... ok
test test_unistd::test_canceling_alarm ... ok
test test_unistd::test_chown ... ok
test test_unistd::test_execve ... ok
test test_unistd::test_fchdir ... ok
test test_unistd::test_fchownat ... ok
test test_unistd::test_fexecve ... ok
test test_unistd::test_fork_and_waitpid ... ok
test test_unistd::test_fpathconf_limited ... ok
test test_unistd::test_ftruncate ... ok
test test_unistd::test_getpid ... ok
test test_unistd::test_getsid ... ok
test_initgroups requires root privileges. Skipping test.
test test_unistd::test_initgroups ... ok
test test_unistd::test_getcwd ... ok
test test_unistd::test_lseek ... ok
test test_unistd::test_mkfifo_directory ... ok
test test_unistd::test_mkstemp ... ok
test test_unistd::test_mkstemp_directory ... ok
test test_unistd::test_mkfifo ... ok
test test_unistd::test_pathconf_limited ... ok
test test_unistd::test_pipe ... ok
test test_unistd::test_pipe2 ... ok
test_setgroups requires root privileges. Skipping test.
test test_unistd::test_setgroups ... ok
test test_unistd::test_sysconf_limited ... ok
test test_unistd::test_sysconf_unsupported ... ok
test test_unistd::test_symlinkat ... ok
test test_unistd::test_wait ... ok
test test_unistd::test_truncate ... ok

test result: ok. 123 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out

     Running target/debug/deps/test_aio_drop-03ba3639a2ac5894

running 1 test
test test_drop ... ok

test result: ok. 1 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out

     Running target/debug/deps/test_lio_listio_resubmit-8496d04c8577e12b

running 1 test
test test_lio_listio_resubmit ... ok

test result: ok. 1 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out

     Running target/debug/deps/test_mount-12230f5546b99452
     Running target/debug/deps/test_ptymaster_drop-895917e279476368

running 1 test
test test_double_close ... ok

test result: ok. 1 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out

   Doc-tests nix

running 75 tests
test src/ifaddrs.rs - ifaddrs::getifaddrs (line 112) ... ok
test src/lib.rs - Error::as_errno (line 117) ... ok
test src/pty.rs - pty::posix_openpt (line 92) ... ok
test src/sys/aio.rs - sys::aio::AioCb<'a>::cancel (line 728) ... ok
test src/sys/aio.rs - sys::aio::AioCb<'a>::error (line 783) ... ok
test src/sys/aio.rs - sys::aio::AioCb<'a>::from_boxed_mut_slice (line 482) ... ok
test src/sys/aio.rs - sys::aio::AioCb<'a>::from_boxed_slice (line 365) ... ok
test src/sys/aio.rs - sys::aio::AioCb<'a>::from_boxed_slice (line 396) ... ok
test src/sys/aio.rs - sys::aio::AioCb<'a>::from_boxed_slice (line 421) ... ok
test src/sys/aio.rs - sys::aio::AioCb<'a>::from_fd (line 228) ... ok
test src/sys/aio.rs - sys::aio::AioCb<'a>::from_mut_slice (line 289) ... ok
test src/sys/aio.rs - sys::aio::AioCb<'a>::from_slice (line 644) ... ok
test src/sys/aio.rs - sys::aio::LioCb<'a>::listio_resubmit (line 1156) ... ok
test src/sys/aio.rs - sys::aio::LioCb<'a>::listio (line 1099) ... ok
test src/sys/aio.rs - sys::aio::aio_suspend (line 981) ... ok
test src/sys/ioctl/bsd.rs - request_code_none (line 41) ... ok
test src/sys/ioctl/mod.rs - ioctl_none (line 269) ... ignored
test src/sys/aio.rs - sys::aio::aio_cancel_all (line 927) ... ok
test src/sys/ioctl/mod.rs - ioctl_none_bad (line 310) ... ignored
test src/sys/ioctl/mod.rs - ioctl_none (line 285) ... ok
test src/sys/ioctl/mod.rs - ioctl_read (line 353) ... ignored
test src/sys/ioctl/mod.rs - ioctl_none_bad (line 318) ... ok
test src/sys/ioctl/mod.rs - ioctl_read_bad (line 390) ... ignored
test src/sys/ioctl/mod.rs - ioctl_read_bad (line 398) ... ok
test src/sys/ioctl/mod.rs - ioctl_read_buf (line 694) ... ignored
test src/sys/ioctl/mod.rs - ioctl_readwrite (line 629) ... ignored
test src/sys/ioctl/mod.rs - ioctl_read (line 361) ... ok
test src/sys/ioctl/mod.rs - ioctl_readwrite_bad (line 665) ... ignored
test src/sys/ioctl/mod.rs - ioctl_readwrite_buf (line 762) ... ignored
test src/sys/ioctl/mod.rs - ioctl_write_buf (line 723) ... ignored
test src/sys/ioctl/mod.rs - ioctl_readwrite (line 637) ... ok
test src/sys/ioctl/mod.rs - ioctl_write_int (line 503) ... ignored
test src/sys/ioctl/mod.rs - ioctl_write_buf (line 731) ... ok
test src/sys/ioctl/mod.rs - ioctl_write_int_bad (line 584) ... ignored
test src/sys/ioctl/mod.rs - ioctl_write_int_bad (line 592) ... ok
test src/sys/ioctl/mod.rs - ioctl_write_int (line 515) ... ok
test src/sys/ioctl/mod.rs - ioctl_write_ptr (line 428) ... ignored
test src/sys/ioctl/mod.rs - ioctl_write_int_bad (line 600) ... ok
test src/sys/ioctl/mod.rs - ioctl_write_ptr_bad (line 464) ... ignored
test src/sys/ioctl/mod.rs - ioctl_write_ptr (line 436) ... ok
test src/sys/ioctl/mod.rs - ioctl_write_ptr_bad (line 472) ... ok
test src/sys/ioctl/mod.rs - sys::ioctl (line 122) ... ok
test src/sys/ioctl/mod.rs - sys::ioctl (line 161) ... ok
test src/sys/ioctl/mod.rs - sys::ioctl (line 172) ... ok
test src/sys/ioctl/mod.rs - sys::ioctl (line 209) ... ok
test src/sys/ioctl/mod.rs - sys::ioctl (line 56) ... ok
test src/sys/ioctl/mod.rs - sys::ioctl (line 66) ... ok
test src/sys/ioctl/mod.rs - sys::ioctl (line 90) ... ok
test src/sys/ptrace/bsd.rs - sys::ptrace::bsd::step (line 123) ... ok
test src/sys/select.rs - sys::select::FdSet::highest (line 49) ... ok
test src/sys/socket/mod.rs - sys::socket::CmsgSpace (line 356) ... ok
test src/sys/socket/mod.rs - sys::socket::setsockopt (line 1104) ... ok
test src/sys/termios.rs - sys::termios (line 103) ... ok
test src/sys/socket/mod.rs - sys::socket::ControlMessage::ScmTimestamp (line 473) ... ok
test src/sys/termios.rs - sys::termios (line 116) ... ok
test src/sys/termios.rs - sys::termios (line 130) ... ok
test src/sys/termios.rs - sys::termios (line 25) ... ok
test src/sys/termios.rs - sys::termios (line 39) ... ok
test src/sys/termios.rs - sys::termios (line 63) ... ok
test src/sys/termios.rs - sys::termios (line 90) ... ignored
test src/sys/termios.rs - sys::termios::Termios::get_libc_termios (line 200) ... ok
test src/sys/termios.rs - sys::termios (line 76) ... ok
test src/unistd.rs - unistd::alarm (line 1500) ... ok
test src/sys/wait.rs - sys::wait::WaitStatus::from_raw (line 173) ... ok
test src/unistd.rs - unistd::close (line 903) ... ok
test src/unistd.rs - unistd::close (line 916) ... ok
test src/unistd.rs - unistd::fork (line 192) ... ok
test src/unistd.rs - unistd::alarm (line 1513) ... ok
test src/unistd.rs - unistd::gethostname (line 874) ... ok
test src/unistd.rs - unistd::initgroups (line 1444) ... ok
test src/unistd.rs - unistd::getcwd (line 544) ... ok
test src/unistd.rs - unistd::mkdir (line 438) ... ok
test src/unistd.rs - unistd::mkfifo (line 481) ... ok
test src/unistd.rs - unistd::setgroups (line 1311) ... ok
test src/unistd.rs - unistd::mkstemp (line 1612) ... ok

test result: ok. 61 passed; 0 failed; 14 ignored; 0 measured; 0 filtered out
```